core/src/components/datetime/datetime.ios.scss

```diff
@@ -72,6 +72,116 @@
+    /**
+     * When the computed font size is 24px
+     * we want the days of the week to only
+     * show the first letter of each day.
+     *
+     * We use the modulus operator to ensure
+     * that any font size less than 24px is
+     * unchanged (i.e. 16 % 24 = 16). If the
+     * font size is 24px then the remainder
+     * will be 0.
+     */
+    $max-font-size: min(1rem, $datetime-dynamic-font-breakpoint);
+    $remainder: mod($max-font-size, $datetime-dynamic-font-breakpoint);
+
+    /**
+     * From here, we scale the remainder to get
+     * a width value larger than 20px.
+     * We then clamp the width to have a max width
+     * of 100% so that each element does not appear
+     * too wide if $width is a very large number.
+     *
+     * When the font size is >=24px then $remainder
+     * will be 0px. This results in $width being 0px
+     * since 0px * 10 is still 0px. We set a minimum
+     * width of 20px below so that the days of the
+     * week still show when the font size is >=24px
+     * and have a fixed width of 20px.
+     */
+    $width: calc($remainder * 10);
+    width: clamp(20px, #{$width}, 100%);
+
+    /**
+     * We show the first letter of the week
+     * by setting the width small such that all other
+     * letters wrap to the next line. While we do have
+     * overflow: hidden to visually hide the overflowing
+     * text, we need to explicitly set the height of the
+     * container too otherwise it will increase as the text wraps.
+     */
+    height: $datetime-dynamic-font-breakpoint;
+
+    /**
+     * The above width will cause
+     * all the other letters to overflow
+     * to other lines. The below code hides
+     * those additional letters so only
+     * the first letter is shown.
+     */
+    overflow: hidden;
+  }
+
+  .calendar-day-highlight {
+    /**
+     * We use the modulus operator to ensure
+     * that any font size less than 24px is
+     * unchanged (i.e. 16 % 24 = 16). If the
+     * font size is 24px then the remainder
+     * will be 0.
+     */
+    $maxFontSize: min(1rem, $datetime-dynamic-font-breakpoint);
+    $remainder: mod($maxFontSize, $datetime-dynamic-font-breakpoint);
+
+    /**
+     * From here, we scale the remainder. The purpose of
+     * scaling the remainder is to get a border radius
+     * large enough such that the element's
+     * shape is a circle.
+     *
+     * Note that when the font size is 24px
+     * then the remainder will be 0, so $radius
+     * will still be zero.
+     */
+    $radius: calc($remainder * 10);
+
+    /**
+     * In the event that $radius is 0
+     * we use max() to ensure a minimum
+     * border radius of 8px which will
+     * give the day button the rounded
+     * rectangle appearance.
+     * For font sizes smaller than 24px
+     * the border radius will give a circle appearance.
+     */
+    @include border-radius(max(8px, $radius));
+  }
+}
+
+/**
+ * iOS <17 has a bug where the result of mod() is
```

Comment:
I'm not observing the day of the week truncating to a single letter when increasing the font size on iOS 16.6.

Should it? I assumed it should based on the [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/CSS/mod#browser_compatibility). But I also see you called out a bug.

Reply:
No it should not. As per the comment, iOS <17 has a bug with `mod()` values so this feature is limited to iOS 17+ for now.